### PR TITLE
feat(annotations): add annotation by URL

### PR DESCRIPTION
**Pull request by kschelonka**
To support Android access patterns and v3-proxy-api.

[POCKET-10026]

[POCKET-10026]: https://mozilla-hub.atlassian.net/browse/POCKET-10026?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: servers/annotations-api/src/dataservices/dataloaders.spec.ts [hunk 1]
```diff
@@ -44,10 +43,7 @@ describe('dataloaders', () => {
       notesStub = jest
         .spyOn(NotesDataService.prototype, 'getMany')
         .mockReturnValue(Promise.resolve(mockNotesResponse));
-      notesLoader = createNotesLoader(dynamo, {
-        isPremium: true,
-        notesService,
-      } as IContext);
+      notesLoader = createNotesLoader(notesService);
```
Comment (kschelonka) [inline servers/annotations-api/src/dataservices/dataloaders.spec.ts:46]: just streamlining injected dependencies here

File: servers/annotations-api/src/models/HighlightsModel.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  Highlight,
+  CreateHighlightInput,
+  HighlightNote,
+  UpdateHighlightInput,
+  SavedItemAnnotations,
+  CreateHighlightByUrlInput,
+} from '../__generated__/resolvers-types';
+import { ParserAPI } from '../dataservices/parserApi';
+
+export class HighlightsModel {
```
Comment (kschelonka) [inline servers/annotations-api/src/models/HighlightsModel.ts:16]: this is just moving the code from `resolvers.ts` into a model for better visibility

File: servers/annotations-api/src/models/HighlightsModel.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  forbidden() {
+    throw new ForbiddenError(
+      'This feature is restricted to premium Pocket accounts',
+    );
+  }
+  async getByItemId(itemId: string): Promise<SavedItemAnnotations> {
+    const highlights = await this.highlightService.getByItemId(itemId);
+    return { highlights };
+  }
+  async createOneByUrl(input: CreateHighlightByUrlInput): Promise<Highlight> {
```
Comment (kschelonka) [inline servers/annotations-api/src/models/HighlightsModel.ts:32]: the only new method

File: servers/annotations-api/src/server/apollo/context.ts [hunk 4]
```diff
@@ -66,15 +79,6 @@ export class ContextManager implements IContext {
 
     return apiId instanceof Array ? apiId[0] : apiId;
   }
-
-  get notesService(): NotesDataService {
```
Comment (kschelonka) [inline servers/annotations-api/src/server/apollo/context.ts:70]: the logic for permissions has been moved to the mdoel

**Review by bassrock — APPROVED** [review]

**Pull request merged**